chain/message_collection_store.go

```diff
@@ -0,0 +1,61 @@
+package chain
+import (
+	"context"
+
+	"github.com/ipfs/go-hamt-ipld"
+	"github.com/ipfs/go-cid"	
+
+	"github.com/filecoin-project/go-filecoin/types"
+)
+
+type MessageCollectionStore struct {
```

Comment:
I suggest just `MessageStore` - we'll very likely want to add access for individual messages later.